### PR TITLE
412 mod introduce check for mute to instantly reject commander vote

### PR DESCRIPTION
**Pull request by GerGotha**
Added a message which tells the player that starts a vote for a certain commander that the target player is muted and cannot be commander.
I have also refactored smaller parts of the command handler to avoid the "You are muted." message whenever you enter a command.

Please test it with a few players before sending it live since I wasn't able to test it that much.

### CODE REVIEW

File: src/Module.Server/CrpgSubModule.cs [hunk 1]
```diff
@@ -16,6 +16,8 @@
 using TaleWorlds.Engine.InputSystem;
 
 using TaleWorlds.MountAndBlade.GameKeyCategory;
+using Crpg.Module.Common.ChatCommands;
```
Comment (Muparadzi) [inline src/Module.Server/CrpgSubModule.cs:19]: Should be under `#if CRPG_SERVER` preprocessor directive (L24)

File: src/Module.Server/Common/ChatCommands/ChatCommandsComponent.cs [hunk 2]
```diff
@@ -13,52 +18,61 @@
...
+                new MuteCommand(this, crpgClient),
+                new BanCommand(this, crpgClient),
+                new MapCommand(this),
+                new HotConstantUpdateCommand(this),
+                new OrderCommand(this),
+                new HelpCommand(this),
         };
 #else
-        _commands = Array.Empty<ChatCommand>();
+            _commands = Array.Empty<ChatCommand>();
```
Comment (Muparadzi) [inline src/Module.Server/Common/ChatCommands/ChatCommandsComponent.cs:53]: whitespace to be removed

File: src/Module.Server/Common/Commander/CrpgCommanderPollComponent.cs [hunk 3]
```diff
@@ -226,6 +232,15 @@ private void OpenCommanderPollOnServer(NetworkCommunicator pollCreatorPeer, Netw
             return;
         }
 
+        CrpgUser? crpgUser = targetPeer.GetComponent<MissionPeer>().GetComponent<CrpgPeer>().User;
+        if (crpgUser?.Restrictions.Any(r => r.Type == CrpgRestrictionType.Chat) ?? false)
```
Comment (Muparadzi) [inline src/Module.Server/Common/Commander/CrpgCommanderPollComponent.cs:236]: We should probably also check whether `targetPeer.IsMuted`. If they are muted during the same game, it is unlikely that the server has fetched the updated crpgUser with the chat restriction.

File: src/Module.Server/Common/Commander/CrpgCommanderPollComponent.cs [hunk 4]
```diff
@@ -7,6 +12,7 @@
 using TaleWorlds.MountAndBlade;
 using TaleWorlds.MountAndBlade.Diamond;
 using TaleWorlds.MountAndBlade.Network.Messages;
+using static TaleWorlds.MountAndBlade.MovementOrder;
```
Comment (Muparadzi) [inline src/Module.Server/Common/Commander/CrpgCommanderPollComponent.cs:15]: unused

**Review by Muparadzi — CHANGES_REQUESTED** [review]
Thanks for returning and helping out! I need to get some players together to test the commander stuff, or if Epic Games update to 1.2.12.

File: src/Module.Server/CrpgSubModule.cs (hunk 1 above)
Comment (GerGotha) [reply src/Module.Server/CrpgSubModule.cs]: move

File: src/Module.Server/Common/ChatCommands/ChatCommandsComponent.cs (hunk 2 above)
Comment (GerGotha) [reply src/Module.Server/Common/ChatCommands/ChatCommandsComponent.cs]: fixed

File: src/Module.Server/Common/Commander/CrpgCommanderPollComponent.cs (hunk 3 above)
Comment (GerGotha) [reply src/Module.Server/Common/Commander/CrpgCommanderPollComponent.cs]: Added the local mute check too.

File: src/Module.Server/Common/Commander/CrpgCommanderPollComponent.cs (hunk 4 above)
Comment (GerGotha) [reply src/Module.Server/Common/Commander/CrpgCommanderPollComponent.cs]: removed

**Pull request merged**